pthread deprecation warnings

```
In file included from c:\mingw\include\pthread.h:76:0,
                 from src/common/tpt-thread.h:16,
                 from src\client\DownloadManager.h:3,
                 from src\client\Download.cpp:3:
c:\mingw\include\_ptw32.h:81:3: error: #warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB." [-Werror=cpp]
 # warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB."
   ^~~~~~~
```
```
In file included from c:\mingw\include\pthread.h:76:0,
                 from src/common/tpt-thread.h:16,
                 from src\client\DownloadManager.h:3,
                 from src\client\DownloadManager.cpp:1:
c:\mingw\include\_ptw32.h:81:3: error: #warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB." [-Werror=cpp]
 # warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB."
   ^~~~~~~
```
```
In file included from c:\mingw\include\pthread.h:76:0,
                 from src/common/tpt-thread.h:16,
                 from src\client/requestbroker/RequestBroker.h:7,
                 from src\client/Client.h:16,
                 from src\PowderToySDL.cpp:46:
c:\mingw\include\_ptw32.h:81:3: error: #warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB." [-Werror=cpp]
 # warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB."
   ^~~~~~~
```
```
In file included from c:\mingw\include\pthread.h:76:0,
                 from src/common/tpt-thread.h:16,
                 from src\client\requestbroker/RequestBroker.h:7,
                 from src\client\Client.h:16,
                 from src\client\Client.cpp:31:
c:\mingw\include\_ptw32.h:81:3: error: #warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB." [-Werror=cpp]
 # warning "PTW32_STATIC_LIB is deprecated; use _PTW32_STATIC_LIB."
   ^~~~~~~
```

--- a/SConscript
+++ b/SConscript
@@ -473,7 +473,7 @@ elif GetOption('release'):
 
 if GetOption('static'):
 	if platform == "Windows":
-		env.Append(CPPDEFINES=['PTW32_STATIC_LIB'])
+		env.Append(CPPDEFINES=['_PTW32_STATIC_LIB'])
 		if msvc:
 			env.Append(CPPDEFINES=['ZLIB_WINAPI'])
 		else:

--- a/src/common/tpt-thread.h
+++ b/src/common/tpt-thread.h
@@ -18,7 +18,7 @@
 
 // Fix deprecation warnings with recent pthread versions on Windows
 #ifdef _PTW32_STATIC_LIB
-#if PTW32_VERSION < 1, 1, 0, 0
+#if PTW32_VERSION =< 2, 8, 0, 0
 #define PTW32_STATIC_LIB
 #endif
 #endif

--- a/src/common/tpt-thread.h
+++ b/src/common/tpt-thread.h
@@ -16,4 +16,11 @@
 #include <pthread.h>
 #undef GetUserName
 
+// Fix deprecation warnings with recent pthread versions on Windows
+#ifdef _PTW32_STATIC
+#if PTW32_VERSION < 1, 1, 0, 0
+#define PTW32_STATIC
+#endif
+#endif
+
 #endif

--- a/src/common/tpt-thread.h
+++ b/src/common/tpt-thread.h
@@ -2,7 +2,7 @@
 #define TPT_THREAD_H
 
 #if defined(WIN) && defined(__GNUC__)
-#define TH_ENTRY_POINT __attribute__((force_align_arg_pointer)) 
+#define TH_ENTRY_POINT __attribute__((force_align_arg_pointer))
 #define _TIMESPEC_DEFINED
 #else
 #define TH_ENTRY_POINT
@@ -17,9 +17,9 @@
 #undef GetUserName
 
 // Fix deprecation warnings with recent pthread versions on Windows
-#ifdef _PTW32_STATIC
+#ifdef _PTW32_STATIC_LIB
 #if PTW32_VERSION < 1, 1, 0, 0
-#define PTW32_STATIC
+#define PTW32_STATIC_LIB
 #endif
 #endif
 

